### PR TITLE
Add support for Python 2.6

### PR DESCRIPTION
**Pull request by tribut**
As mentioned in https://github.com/mmb/weechat-otr/pull/86 the changes for Python 2.6 compatibility are unfortunately a little bit more complicated than I initially thought. The most intrusive thing is still having to use `{n}` instead of `{}` in format strings, but I think that's fine.
Note that this patches potr before running the tests on travis, see https://github.com/python-otr/pure-python-otr/pull/55.

### CODE REVIEW

**Comment by mmb** [comment]
Looks good :+1:

**Comment by tribut** [comment]
Almost all done. Please have a look at https://github.com/tribut/weechat-otr/commit/c591637c3718a1ec7b71cd581810550e43de4018#commitcomment-9257862 and tell me what you think. We should probably change our tests instead of that hack.

**Comment by tribut** [comment]
I've added a commit mentioning the yet-to-be-released version of potr as requirement. When the new version of potr hits pip, this should be ready to merge (I think).

**Comment by Mikaela** [comment]
> I've added a commit mentioning the yet-to-be-released version of potr as requirement. When the new version of potr hits pip, this should be ready to merge (I think).

You should probably also update requirements.txt.

(It's also possible to specify git repos there, but that might be too experimental :))

**Comment by Mikaela** [comment]
This was just wondered at #otr at OFTC, is there anything new of that version of potr? I think it might already be released.

**Comment by tribut** [comment]
I've rebased this onto master, but we still kind of have a dependency on Python 2.7 via https://github.com/python-otr/pure-python-otr/issues/74. @koolfy? 1.0.2? 🤡 

Otherwise I would say merge this now. It's not making anything worse and we can always drop https://github.com/mmb/weechat-otr/pull/99/commits/45e3b380d1bd9a456ef2bccd091ff2f851c6ad68 later.

File: weechat_otr_test/mock_weechat.py [hunk 1]
```diff
@@ -113,14 +115,14 @@ def restore(self):
 
     def snapshot_weechat_dir(self):
         tar_io = io.BytesIO()
-        with tarfile.open(fileobj=tar_io, mode='w') as tar:
+        with closing(tarfile.open(fileobj=tar_io, mode='w')) as tar:
```
Comment (mmb) [inline weechat_otr_test/mock_weechat.py]: If we have to pull in another module just to use `with` it seems better just to do the old `tarfile.open` ... `tar.close()`.

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 python:
+  - "2.6"
   - "2.7"
   - "3.3"
   - "3.4"
   - "3.5"
 install:
   - "pip install pylint"
   - "pip install -r requirements.txt"
+  - "if [ $TRAVIS_PYTHON_VERSION == '2.6' ]; then pip install unittest2; fi"
```
Comment (mmb) [inline .travis.yml]: Is it only test discovery that doesn't work in 2.6?

If it's only that maybe we can find a way to do discovery without pulling in unittest2.

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+  - "sed -i \"s/'ascii', errors='replace'/'ascii', 'replace'/\" $(python -c 'import distutils.sysconfig as s; print(s.get_python_lib())')/potr/proto.py"
 script: ./run_tests.sh
```
Comment (mmb) [inline .travis.yml]: We can start merging the smaller changes like `{}` to `{0}` to get closer to 2.6 support but I think we should wait until potr supports it before merging the more 2.6-specific changes. I don't want the code to give the impression that it's usable with 2.6 if it isn't yet.

File: weechat_otr_test/test_command_finish.py [hunk 4]
```diff
@@ -21,23 +21,23 @@ def after_setup(self):
         weechat_otr.ACCOUNTS['nick@server'] = account
 
     def test_finish_buffer(self):
-        weechat_otr.command_cb(None, 'server_nick2_buffer', 'finish')
+        self.call_command_cb(None, 'server_nick2_buffer', 'finish')
```
Comment (mmb) [inline weechat_otr_test/test_command_finish.py]: These I would consider a bug in the current tests unrelated to 2.6 and could potentially be their own PR. They should have been using `weechat_otr.PYVER.to_str` like the other tests. 

How do you feel about calling to_str in each test instead of using a helper? It's less DRY but I think it makes what the test is doing clearer.

**Review by mmb — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: weechat_otr_test/mock_weechat.py (hunk 1 above)
Comment (tribut) [inline weechat_otr_test/mock_weechat.py]: You're right. I've changed that and put all the "minor"/style changes into another PR, as you suggested: https://github.com/mmb/weechat-otr/pull/134

File: weechat_otr_test/test_command_finish.py (hunk 4 above)
Comment (tribut) [inline weechat_otr_test/test_command_finish.py]: Hmm... maybe you're right. I'm not sure.

It feels wrong to have to decide whether or not to call `to_str` based on the contents of the string. And adding it to every string is very verbose. But adding our own `assert*` functions with magic string conversion is probably just as bad. So yeah, I'll do that after #134.

Comment (mmb) [inline weechat_otr_test/test_command_finish.py]: I think we should always call `to_str` when we call something that WeeChat calls in a test because WeeChat sends bytes.

Our strategy has been to convert bytes to unicode as soon as they come into our code and convert them back to bytes only on the way out if calling something that needs bytes.

Comment (tribut) [inline weechat_otr_test/test_command_finish.py]: Of course, that is the correct behavior. I'm just looking for a way to not have 100+ (if I counted correctly) calls to `weechat_otr.PYVER.to_str` in the tests which are easy to forget 😏 

How would you feel about importing a derived class of `weechat_otr` in the tests that only contains all the `*_cb` methods which convert the strings to bytes and calls the parent method?
But if you just want all the strings converted in the argument lists, feel free to say so.

**Comment by tribut** [comment]
Python 2 is pretty much EOL everywhere now, so I am closing this.
